tests/integration/ambient/baseline_test.go

```diff
@@ -631,9 +631,10 @@ kind: EnvoyFilter
 metadata:
   name: inbound
 spec:
-  workloadSelector:
-    labels:
-      gateway.networking.k8s.io/gateway-name: "{{.Destination}}"
+  targetRefs:
```

Comment:
This's a breaking change, but we should use `targetRefs` with waypoint, right?
cc @howardjohn

Reply:
I think the original PR has a test showing the label selector doesn't work as well as that targetRef does right? 

Re breaking - it is. And that's good. EnvoyFilter will break again in 1.27

Reply:
IIUC, https://github.com/istio/istio/pull/55149 skipped when targetRefs is not nil, which means not implement.

Yeah, The goal is change the context to WAYPOINT in v1.27.
My thought is that, If we plan to backport this fix to 1.25/1.26, breaking is not a good idea.

Reply:
EF in ambient has never been supported.. breaking wouldn't be good, but it's a feature we never offered, either. I'd flag the breaking change in the first comment of the PR so the RMs have a better chance of seeing it should it be backported.

Reply:
So I think I should take the way of the right thing(targetRefs is the right way) and leave the context to https://github.com/istio/api/pull/3442.

At the end, we will expect that user use targetRefs and WAYPOINT context in the `1.27`.